x/superfluid/keeper/stake_test.go

```diff
@@ -1028,6 +1031,522 @@ func (s *KeeperTestSuite) TestRefreshIntermediaryDelegationAmounts() {
 	}
 }
 
+func (s *KeeperTestSuite) TestUnbondConvertAndStake() {
+	defaultJoinTime := s.Ctx.BlockTime()
+	type tc struct {
+		superfluidUndelegating bool
+		unlocking              bool
+		unlocked               bool
+
+		expectedError bool
+	}
+	testCases := map[string]tc{
+		"lock that is superfluid delegated": {},
+		"lock that is superfluid undelegating": {
+			unlocking:              true,
+			superfluidUndelegating: true,
+		},
+		"lock that is unlocking": {
+			unlocking:              true,
+			superfluidUndelegating: false,
+		},
+		"unlocked gamm shares": {
+			unlocking:              true,
+			superfluidUndelegating: false,
+		},
```

Comment:
Good catch, added and modified test case so that it tests both 
- unlocking case
- unlocked case
in https://github.com/osmosis-labs/osmosis/pull/5949/commits/c89e9c3945e0dc402e836c4805182c1718d81be5